File: app/src/main/java/org/dhis2/usescases/org_unit_selector/OrgUnitSelectorHolder.java
package org.dhis2.usescases.org_unit_selector;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.dhis2.databinding.ItemOuTreeBinding;
import org.dhis2.utils.filters.FilterManager;
import org.hisp.dhis.android.core.organisationunit.OrganisationUnit;

class OrgUnitSelectorHolder extends RecyclerView.ViewHolder {
    private final ItemOuTreeBinding binding;

    public OrgUnitSelectorHolder(@NonNull ItemOuTreeBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(OrganisationUnit organisationUnit) {
        binding.ouName.setText(organisationUnit.displayName());
        ViewGroup.MarginLayoutParams marginParams = (ViewGroup.MarginLayoutParams) binding.icon.getLayoutParams();

        marginParams.leftMargin += (organisationUnit.level() - 1) * 25;

        binding.checkBox.setChecked(FilterManager.getInstance().getOrgUnitFilters().contains(organisationUnit));
        binding.checkBox.setOnCheckedChangeListener((compoundButton, b) ->
                FilterManager.getInstance().addOrgUnit(organisationUnit));
    }
}

File: app/src/main/java/org/dhis2/usescases/orgunitselector/OrgUnitSelectorHolder.kt
package org.dhis2.usescases.orgunitselector

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import org.dhis2.R
import org.dhis2.commons.filters.FilterManager
import org.dhis2.databinding.ItemOuTreeBinding

internal class OrgUnitSelectorHolder(private val binding: ItemOuTreeBinding) :
    RecyclerView.ViewHolder(binding.root) {

    private lateinit var node: TreeNode

    fun bind(
        node: TreeNode
    ) {
        this.node = node
        binding.checkBox.setOnCheckedChangeListener(null)

        binding.ouName.text = node.content.displayName()
        node.isChecked = FilterManager.getInstance().exist(node.content)
        val marginParams = binding.root.layoutParams as ViewGroup.MarginLayoutParams
        marginParams.leftMargin = (node.level - 1) * 40
        binding.checkBox.isChecked = node.isChecked

        if (!node.hasChild) {
            binding.icon.setImageResource(R.drawable.ic_circle_primary)
        } else if (node.isOpen) {
            binding.icon.setImageResource(R.drawable.ic_remove_circle_primary)
        } else {
            binding.icon.setImageResource(R.drawable.ic_add_circle)
        }

        binding.checkBox.setOnCheckedChangeListener { _, isChecked ->
            FilterManager.getInstance().addIfCan(node.content, isChecked)
        }
    }
}
